redbot/cogs/mod/settings.py

```diff
@@ -283,33 +295,93 @@ async def mentionspam_ban(self, ctx: commands.Context, max_mentions: int):
 
     @modset.command()
     @commands.guild_only()
-    async def deleterepeats(self, ctx: commands.Context, repeats: int = None):
+    async def deleterepeats(
+        self, ctx: commands.Context, repeats: int = None, for_channel: str = None
+    ):
         """Enable auto-deletion of repeated messages.
 
         Must be between 2 and 20.
 
         Set to -1 to disable this feature.
+
+        Add "this" if you want to specify for only one channel
```

Comment:
The argument to specify the setting should only apply for a channel should be an optional `discord.TextChannel` (or perhaps a greedy one if we want to support passing multiple channels?). It should not be a string, and it should not lock the user into only picking the current channel.